### ISSUE TITLE
[bug] `use_sliding_window` doesn't work as expected

### ISSUE THREAD

**Issue by ZhiyuLi-Nvidia** [huggingface/transformers#38002]
### System Info

* transformer: main
* pytorch, cuda: anyversion

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```
from transformers import AutoModelForCausalLM
import numpy as np
import torch


MODEL_NAME = "deepseek-ai/DeepSeek-R1-Distill-Qwen-1.5B"
model = AutoModelForCausalLM.from_pretrained(MODEL_NAME)

input_ids = torch.randint(0, 100, (1, 8192))

with torch.no_grad():
    output_raw = model(input_ids)

# correct situaion: should be the same as the original model since use_sliding_window is False
model_no_sliding = AutoModelForCausalLM.from_pretrained(MODEL_NAME, sliding_window=None)

with torch.no_grad():
    output_non_sliding = model_no_sliding(input_ids)


np.testing.assert_allclose(output_raw.logits[:, :4096], output_non_sliding.logits[:, :4096])

# wrong: the logits are unexpectedly different with sliding_window=4096
np.testing.assert_allclose(output_raw.logits[:, 4096:], output_non_sliding.logits[:, 4096:])
```

### Expected behavior

# description
What is expected:   

```
  "sliding_window": 4096,
  "use_sliding_window": false,
```
`use_sliding_window` is set as false in `deepseek-ai/DeepSeek-R1-Distill-Qwen-1.5B` [here](https://huggingface.co/deepseek-ai/DeepSeek-R1-Distill-Qwen-1.5B/blob/main/config.json#L20). We do expect sliding window is disabled. In other words, we should expect the same results even with different `sliding_window`.

However, the results are different in the repro script.

# Root cause
Attention Mask is changed according to `sliding_window` without respect on `use_sliding_window`.

https://github.com/huggingface/transformers/blob/3c0796aaeac722152836ea063d542a4c628a75be/src/transformers/models/qwen2/modeling_qwen2.py#L708-L715

If we add some printing under this conditional block, we can clearly see attention mask is changed even with `use_sliding_window=false`

**Comment by Rocketknight1** [comment]
Not sure who's the right code owner here - cc @gante @zucchini-nlp, but feel free to tag Arthur instead if you think he's more appropriate!

**Comment by zucchini-nlp** [comment]
Nice catch! Indeed I think we should be checking both values, given that in attention layer we pass sliding window only when `use_sliding_window`. I can open a PR for that 🤗
